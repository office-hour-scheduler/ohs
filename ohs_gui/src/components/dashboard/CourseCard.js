import React, { Component } from 'react';
import { Link } from 'react-router-dom'

class CourseCard extends Component {
  render() {
    return (
      <Link to={`/course/${this.props.course}`}>
        <div className="course card-element">
          {this.props.course}
        </div>
      </Link>
    );
  }
}

export default CourseCard;
