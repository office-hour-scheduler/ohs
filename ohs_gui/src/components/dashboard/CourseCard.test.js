import React from 'react';
import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';

import CourseCard from './CourseCard';

const COURSE = 'CSC302';

it('should render correctly with no props', () => {
  const t = () => {
    shallow(<CourseCard />);
  };
  expect(t).toThrow(Error);
});

it('renders course name', () => {
  const component = shallow(<CourseCard course={COURSE} />);
  const text = component.find('.course').text();
  expect(text).toEqual(COURSE);
});

it('check if it renders properly', () => {
  const component = shallow(<CourseCard course={COURSE} />);
  expect(component).toMatchSnapshot();
});

it('test if Link is created to the right URL', () => {
  const component = shallow(<CourseCard course={COURSE} />);
  const link = component.find(Link).prop('to');
  expect(link).toEqual(`/course/${COURSE}`);
});
