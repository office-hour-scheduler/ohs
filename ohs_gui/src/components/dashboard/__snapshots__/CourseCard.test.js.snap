// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if it renders properly 1`] = `
<Link
  replace={false}
  to="/course/CSC302"
>
  <div
    className="course card-element"
  >
    CSC302
  </div>
</Link>
`;